test/performance/operf.go

```diff
@@ -0,0 +1,666 @@
+	MemQueryTemplatePod       = `sum(container_memory_working_set_bytes{cluster="", namespace="{{.Namespace}}",container!="", image!=""} * on(namespace,pod) group_left(workload, workload_type) namespace_workload_pod:kube_pod_owner:relabel{cluster="", namespace="{{.Namespace}}", workload="{{.Name}}", workload_type="{{.WorkloadType}}"}) by (pod)`
+	MemQueryTemplateContainer = `sum(container_memory_working_set_bytes{cluster="", namespace="{{.Namespace}}",container!="", image!=""} * on(namespace,pod) group_left(workload, workload_type) namespace_workload_pod:kube_pod_owner:relabel{cluster="", namespace="{{.Namespace}}", workload="{{.Name}}", workload_type="{{.WorkloadType}}"}) by (container)`
+	Mebibyte                  = 1048576 // bytes
+)
+
+const (
+	UsageHelp = `Usage of operf:
+This script provides LVM operator metrics retrieval (for all the units in the operator):
+
+- PVCs creation and usage:
+	It creates <instances> of "busybox" pods using PVCs provisioned via the Storage class provided
+	Example:
+	# go run test/performance/operf.go -token sha256~cj81ClyUYu7g05y8K-uLWm2AbrKTbNEQ96hEJcWStQo -instances 16
+
+Parameters:`
+)
+
+var (
+	// flags
+	Instances            *int
+	TestNamespace        *string
+	TestStorageClassName *string
+	NamePattern          *string
+	Token                *string
+
+	// variables
+	PrometheusURL string
+	PVCLabels     = map[string]string{"app": "testPerformance"}
+	Ctx           = context.Background()
+)
+
+func main() {
+	Instances = flag.Int("instances", 4, "Number of Pods/Pvcs (each Pod uses a different PVC) to create in the test")
+	TestNamespace = flag.String("namespace", "lvm-operator-system", "Namespace where operator is deployed and the PVCs and test pods will be deployed/undeployed")
+	TestStorageClassName = flag.String("sc", "odf-lvm-vg1", "Name of the topolvm storage class that will be used in the PVCs")
+	NamePattern = flag.String("pattern", "perfotest", "Pattern used to build the PVCs/Pods names")
+	Token = flag.String("token", "", "Mandatory authentication token needed to connect with the Openshift cluster")
+
+	flag.Usage = func() {
+		writer := tabwriter.NewWriter(os.Stdout, 0, 8, 1, '\t', tabwriter.AlignRight)
+		flagSet := flag.CommandLine
+		fmt.Println(UsageHelp)
+		order := []string{"instances", "namespace", "pattern", "sc", "token"}
+		for _, name := range order {
+			flag := flagSet.Lookup(name)
+			fmt.Fprintf(writer, "\t-%s\t(default: %s)\t%s\n", flag.Name, flag.DefValue, flag.Usage)
+		}
+		writer.Flush()
+	}
+
+	flag.Parse()
+
+	if *Token == "" {
+		panic("Please provide the token parameter to connect with the Openshift cluster. Aborting test.")
```

Comment:
panics are usually for errors - it is better to just exit here with an error message.

Reply:
do you mean

```
fmt.Println("Please provide the token parameter to connect with the Openshift cluster. Aborting test.")
os.Exit(1)
```

Reply:
Yes.

Reply:
done